scripts/check_formatting

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Runs clang-format and rustfmt on the entire codebase to check if it's 
+# well formatted. Exit codes:
+#  - 1 there are files to be formatted.
+#  - 0 everything looks fine.
+
+# We ask clang-format to generate an XML output with the deisred formattings.
+# We check if any "replacement" tag exists in the output.
+CLANG_FORMAT_EXIT_CODE=0
+find oak examples -type f -name '*.h' -o -name '*.cc' -o -name '*.proto' \
```

Comment:
In order for this to actually work properly, you need to enclose this in parentheses. [Source](https://github.com/koalaman/shellcheck/wiki/SC2146). Like:

```shell
find oak examples \( -type f -name '*.h' -o -name '*.cc' -o -name '*.proto' \)
```

Reply:
Done.